src/bin/vip-logs.js

```diff
@@ -43,14 +42,89 @@ export async function getLogs( arg: string[], opt ): Promise<void> {
 
 	await trackEvent( 'logs_command_success', {
 		...trackingParams,
-		logs_output: logs.length,
+		total: logs.nodes.length,
 	} );
 
-	if ( ! logs.length ) {
+	if ( ! logs.nodes.length ) {
 		console.error( 'No logs found' );
 		return;
 	}
 
+	printLogs( logs.nodes, opt.format );
+}
+
+export async function followLogs( opt ): Promise<void> {
+	let after = null;
+	let isFirstRequest = true;
+	// How many times have we polled?
+	let requestNumber = 0;
+
+	const trackingParams = getBaseTrackingParams( opt );
+
+	// Set an initial default delay
+	let delay = DEFAULT_POLLING_DELAY_IN_SECONDS;
+
+	while ( true ) {
+		const limit = isFirstRequest ? opt.limit : LIMIT_MAX;
+
+		requestNumber++;
+		trackingParams.request_number = requestNumber;
+		trackingParams.request_delay = delay;
+		trackingParams.limit = limit;
+
+		let logs;
+		try {
+			logs = await logsLib.getRecentLogs( opt.app.id, opt.env.id, opt.type, limit, after );
+
+			await trackEvent( 'logs_command_follow_success', {
+				...trackingParams,
+				total: logs?.nodes.length,
+			} );
+		} catch ( error ) {
+			await trackEvent( 'logs_command_follow_error', { ...trackingParams, error: error.message } );
+
+			// If the first request fails we don't want to retry (it's probably not recoverable)
+			if ( isFirstRequest ) {
+				console.error( 'Error fetching initial logs' );
```

Comment:
[1] Shouldn't we print the error message? If the user is hitting a rate limit for example, they will never know.
```suggestion
				console.error( `Error fetching initial logs: ${ error.message }` );
```

Reply:
I'm not sure about this... In the end this will always be an error coming from the Apollo client, and it manages errors with the `onError` handler. And we are already printing there all the errors of the GraphQL request. So this `console.error()` (and the one below) should always be printed after the actual errors. I'll try it anyway, forcing some errors, to see what we would obtain here.

Reply:
oh, if the errors are already being printed, please just ignore this suggestion :upside_down_face:

Reply:
Yep, I confirm we are printing at least the same as `error.message` (and more if we have multiple errors), so this is not needed 😊  But thanks for the suggestion anyway! It made sense without knowing the context